diff -rupN linux.orig/arch/arm64/boot/dts/rockchip/rk3568-evb-rgb30.dtsi linux/arch/arm64/boot/dts/rockchip/rk3568-evb-rgb30.dtsi
--- linux.orig/arch/arm64/boot/dts/rockchip/rk3568-evb-rgb30.dtsi	2023-09-16 13:43:01.698697805 +0000
+++ linux/arch/arm64/boot/dts/rockchip/rk3568-evb-rgb30.dtsi	2023-09-16 13:46:47.282148978 +0000
@@ -96,7 +96,7 @@
 			232 233 234 235 236 237 238 239
 			240 241 242 243 244 245 246 247
 			248 249 250 251 252 253 254 255>;
-		default-brightness-level = <77>;
+		default-brightness-level = <16>;
 	};
 
 	charge-animation {
